process/block/sovereign/incomingHeader/extendedHeaderProcessor.go

```diff
@@ -5,40 +5,85 @@ import (
 	"github.com/multiversx/mx-chain-core-go/data"
 	"github.com/multiversx/mx-chain-core-go/data/block"
 	"github.com/multiversx/mx-chain-core-go/data/sovereign"
+	sovDto "github.com/multiversx/mx-chain-core-go/data/sovereign/dto"
 	"github.com/multiversx/mx-chain-core-go/data/transaction"
 	"github.com/multiversx/mx-chain-core-go/hashing"
 	"github.com/multiversx/mx-chain-core-go/marshal"
 
 	"github.com/multiversx/mx-chain-go/process"
 	"github.com/multiversx/mx-chain-go/process/block/sovereign/incomingHeader/dto"
+	"github.com/multiversx/mx-chain-go/process/block/sovereign/incomingHeader/extendedHeader"
 )
 
 type extendedHeaderProcessor struct {
 	headersPool HeadersPool
 	txPool      TransactionPool
 	marshaller  marshal.Marshalizer
 	hasher      hashing.Hasher
+	container   EmptyExtendedHeaderCreatorsContainerHandler
 }
 
-func createExtendedHeader(incomingHeader sovereign.IncomingHeaderHandler, scrs []*dto.SCRInfo) (*block.ShardHeaderExtended, error) {
-	headerV2, castOk := incomingHeader.GetHeaderHandler().(*block.HeaderV2)
-	if !castOk {
-		return nil, errInvalidHeaderType
+func newExtendedHeaderProcessor(
+	headersPool HeadersPool,
+	txPool TransactionPool,
+	marshaller marshal.Marshalizer,
+	hasher hashing.Hasher,
+) (*extendedHeaderProcessor, error) {
+	container := extendedHeader.NewEmptyBlockCreatorsContainer()
```

Comment:
I think this container should be created in sovereignNodeRunner based on a config

Reply:
We can't create it based on any config, it's impossible.
You can also check how we did it with metaBlock, headerV1, headerV2, it's the same flow